pkg/patterns/declarative/pkg/applier/direct.go

```diff
@@ -76,13 +77,14 @@ func (d *DirectApplier) Apply(ctx context.Context, opt ApplierOptions) error {
 	b := d.inner.NewBuilder(opt)
 	f := d.inner.NewFactory(opt)
 
+	dynamicClient, err := f.DynamicClient()
+	if err != nil {
+		return err
+	}
+
 	if opt.Validate {
 		// This potentially causes redundant work, but validation isn't the common path
```

Comment:
I wonder if this redundant work is comment is related to the creation of the DynamicClient. If so this comment should be deleted I guess. Also, is it fine to create that client every time this runs or should we be smarter about it?

Reply:
I think I wrote the original comment.  And yes, I think we're constructing new clients and probably doing discovery again, but in general I thought it wasn't as critical to optimize given that validation isn't as useful as it once was - the server should be doing the same validation here, we don't expect to fail validation given that we're probably constraining the manifests etc.

Maybe we should update the comment to say something like "validation likely makes redundant apiserver requests and is less optimized than the non-validation case, but validation isn't the common path" ?

Reply:
Understood! Sounds good to me!